src/video/wayland/SDL_waylandevents.c

```diff
@@ -1557,7 +1559,12 @@ text_input_done(void *data,
                 struct zwp_text_input_v3 *zwp_text_input_v3,
                 uint32_t serial)
 {
-    /* No-op */
+    SDL_WaylandTextInput *text_input = data;
+    if (!text_input->has_preedit) {
+        char buf[1] = {'\0'};
+        SDL_SendEditingText(buf, 0, 0);
```

Comment:
Looked around the rest of the source, turns out other backends do it like this:

```suggestion
        SDL_SendEditingText("", 0, 0);
```